rolepermissions/roles.py

```diff
@@ -49,24 +49,30 @@ def get_name(cls):
     @classmethod
     def assign_role_to_user(cls, user):
         """
-        Deletes all of user's previous roles, and removes all permissions
-        mentioned in their available_permissions property.
+        Assign this role to a user.
 
         :returns: :py:class:`django.contrib.auth.models.Group` The group for the
             new role.
         """
-        from rolepermissions.shortcuts import remove_role
-
-        remove_role(user)
-
-        group, created = Group.objects.get_or_create(name=cls.get_name())
+        group, _created = Group.objects.get_or_create(name=cls.get_name())
         user.groups.add(group)
         permissions_to_add = cls.get_default_true_permissions()
         user.user_permissions.add(*permissions_to_add)
 
         return group
 
     @classmethod
+    def remove_role_from_user(cls, user):
```

Comment:
I think that by doing this we are doing our best. Since there's no case where we are ADDING a permission it's also a safe approach. In the worst case scenario someone will need to contact support and tell that lost X permission.